scripts/microshift-cleanup-data.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+SCRIPT_NAME=$(basename $0)
+# The 'openshift-ovn-kubernetes' namespace must be in the end of the list
+# to allow for pod deletion when MicroShift is stopped
+PODS_NS_LIST=(openshift-service-ca openshift-ingress openshift-dns openshift-storage openshift-ovn-kubernetes)
+FULL_CLEAN=true
+
+function usage() {
+    echo "Stop all MicroShift services, also cleaning their data"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} <--all | --ovn>"
+    echo "   --all   Clean all MicroShift and OVN data"
+    echo "   --ovn   Clean OVN data only"
+    exit 1
+}
+
+# Parse command line
+case $1 in
+--all)
+    ;;
+--ovn)
+    FULL_CLEAN=false
+    ;;
+*)
+    usage
+    ;;
+esac
+
+# Exit if the current user is not 'root'
+if [ $(id -u) -ne 0 ] ; then
+    echo "The '${SCRIPT_NAME}' script must be run with the 'root' user privileges"
+    exit 0
+fi
+
+if ${FULL_CLEAN} ; then
+    echo "DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?"
+    select yn in "Yes" "No"; do
+        case "${yn}" in
+        Yes) 
+            break
+            ;;
+        *)
+            echo "Aborting cleanup" 
+            exit 0
+            ;;
+        esac
+    done
+fi
+
+echo Stopping MicroShift services
+for service in microshift microshift-etcd ; do
+    systemctl stop --now   ${service} 2>/dev/null
+    systemctl reset-failed ${service} 2>/dev/null
+done
+
+if ${FULL_CLEAN} ; then
+    echo Disabling MicroShift services
+    systemctl disable microshift 2>/dev/null
+fi
+pkill -9 --exact microshift
+pkill -9 --exact microshift-etcd
+
+if ${FULL_CLEAN} ; then
+    echo Removing crio container and image storage
+    crio wipe -f &>/dev/null
+    systemctl restart crio
+else
+    echo Removing MicroShift pods
```

Comment:
Why do we need to kill the pods? Is this part of the greenboot handling, or is that for some other use?

Reply:
It is necessary to kill the pods (ovn last) to allow for further killing of conmon, etc. processes. Otherwise, they hold the /var/run/ovn files open and it's not possible to clean up ovn databases.

Reply:
The script will be more maintainable if we include that sort of information in comments.